src/endpoint/s3/s3_rest.js

```diff
@@ -315,7 +315,7 @@ function get_bucket_and_key(req) {
     return {
         bucket,
         // decode and replace hadoop _$folder$ in key
-        key: decodeURIComponent(key).replace(/_\$folder\$/, '/'),
+        key: decodeURIComponent(key.replace('%', '%25')).replace(/_\$folder\$/, '/'),
```

Comment:
Not sure about that fix. It looks like something that will hunt us in the future.
It usually not a good practice to just replace characters in a request.

Reply:
fixed, see #5814 pull request.